fission/src/ui/panels/configuring/scoring/ZoneConfigPanel.tsx

```diff
@@ -1,84 +1,202 @@
+import SelectButton from "@/ui/components/SelectButton"
+import Jolt from "@barclah/jolt-physics"
+import TransformGizmos from "@/ui/components/TransformGizmos"
+import * as THREE from "three"
+import World from "@/systems/World"
+import { ReactRgbaColor_ThreeColor } from "@/util/TypeConversions"
+import { useTheme } from "@/ui/ThemeContext"
+import MirabufSceneObject, { RigidNodeAssociate } from "@/mirabuf/MirabufSceneObject"
+import { MiraType } from "@/mirabuf/MirabufLoader"
+import { ToggleButton, ToggleButtonGroup } from "@/ui/components/ToggleButtonGroup"
 
 const ZoneConfigPanel: React.FC<PanelPropsImpl> = ({ panelId, openLocation, sidePadding }) => {
-    // somehow get and store which zone is being edited
-    // maybe a global ConfigProvider in App.tsx?
-    // then set all default values to the state of the zone
-    const [, setName] = useState<string>("")
-    const [alliance, setAlliance] = useState<"red" | "blue">("blue")
-    const [, setParent] = useState<string>("")
-    const [, setPoints] = useState<number>(1)
-    const [, setDestroy] = useState<boolean>(false)
-    const [, setPersistent] = useState<boolean>(false)
-    const [, setScale] = useState<[number, number, number]>([1, 1, 1])
+    const { openPanel, closePanel } = usePanelControlContext()
+
+    const transformGizmoRef = useRef<TransformGizmos>()
+
+    const [name, setName] = useState<string>(SelectedZone.zone.name)
+    const [alliance, setAlliance] = useState<"red" | "blue">(SelectedZone.zone.alliance)
+    const [selectedNode, setSelectedNode] = useState<string | undefined>(SelectedZone.zone.parentNode)
+    const [points, setPoints] = useState<number>(SelectedZone.zone.points)
+    const [destroy, setDestroy] = useState<boolean>(SelectedZone.zone.destroyGamepiece)
+    const [persistent, setPersistent] = useState<boolean>(SelectedZone.zone.persistentPoints)
+
+    const [transformMode, setTransformMode] = useState<"translate" | "rotate" | "scale">("translate")
+
+    const { currentTheme, themes } = useTheme()
+    const theme = useMemo(() => {
+        return themes[currentTheme]
+    }, [currentTheme, themes])
+
+    const field = useMemo(() => {
+        const assemblies = [...World.SceneRenderer.sceneObjects.values()]
+        for (let i = 0; i < assemblies.length; i++) {
+            const assembly = assemblies[i]
+            if (!(assembly instanceof MirabufSceneObject)) continue
+
+            if ((assembly as MirabufSceneObject).miraType != MiraType.FIELD) continue
+
+            return assembly
+        }
+
+        return undefined
+    }, [])
+
+    useEffect(() => {
+        closePanel("scoring-zones")
+        configureGizmo()
+    }, [])
+
+    const configureGizmo = () => {
+        transformGizmoRef.current = new TransformGizmos(
+            new THREE.Mesh(
+                new THREE.BoxGeometry(1, 1, 1),
+                World.SceneRenderer.CreateToonMaterial(ReactRgbaColor_ThreeColor(theme.HighlightHover.color))
```

Comment:
I'd look at the translucent materials that Dhruv did in MirabufInstance, and add it to the CreateToonMaterial function. Then prolly adjust the highlight hover color to be like half or 3/4ths opacity.